orc8r/cloud/go/services/metricsd/servicers/servicer_test.go

```diff
@@ -202,6 +202,59 @@ func TestCollect(t *testing.T) {
 	assert.Equal(t, prevInfoLines+1, glog.Stats.Info.Lines())
 }
 
+func TestCollectMismatchedGateway(t *testing.T) {
+	device_test_init.StartTestService(t)
+	configurator_test_init.StartTestService(t)
+	_ = serde.RegisterSerdes(serde.NewBinarySerde(device.SerdeDomain, orc8r.AccessGatewayRecordType, &models.GatewayDevice{}))
+
+	e := &testMetricExporter{}
+	test_init.StartNewTestExporter(t, e)
+	srv := servicers.NewMetricsControllerServer()
+
+	// Create test network
+	networkID := "metricsd_servicer_test_network"
+	test_utils.RegisterNetwork(t, networkID, "Test Network Name")
+
+	// Register a fake gateway
+	gatewayID := "2876171d-bf38-4254-b4da-71a713952904"
+	id := protos.NewGatewayIdentity(gatewayID, "testNwId", "testLogicalId")
+	ctx := id.NewContextWithIdentity(context.Background())
+	test_utils.RegisterGateway(t, networkID, gatewayID, &models.GatewayDevice{HardwareID: gatewayID})
+
+	// Mismatched gateway ID
+	mismatchgatewayID := "2876171d-bf38-4254-b4da-71a713954029"
+
+	name := strconv.Itoa(int(MetricName))
+	key := strconv.Itoa(int(LabelName))
+	value := LabelValue
+	float := 1.0
+	counter_type := dto.MetricType_COUNTER
+	counters := protos.MetricsContainer{
+		GatewayId: mismatchgatewayID,
+		Family: []*dto.MetricFamily{{
+			Type: &counter_type,
+			Name: &name,
+			Metric: []*dto.Metric{
+				{
+					Label: []*dto.LabelPair{{
+						Name:  &key,
+						Value: &value}},
+					Counter: &dto.Counter{
+						Value: &float},
+				}}}}}
```

Comment:
```
protos.MC{
  GatewayID: ...,
  Family: []MF{
    Metric: []M{
      {
        Label: []LP{{Name: , Value:}},
        Counter: {},
      },
    },
  },
}
```

Reply:
Submitted with changes to indentation for better readability.